pkg/controller/flinkapplication/flink_state_machine.go

```diff
@@ -162,16 +162,13 @@ func (s *FlinkStateMachine) handle(ctx context.Context, application *v1alpha1.Fl
 			appErr = s.handleApplicationDeleting(ctx, application)
 		}
 
-		if appErr != nil {
+		if !v1alpha1.IsRunningPhase(application.Status.Phase) {
```

Comment:
You will have to cache phase as well in local variable.

phase := application.Status.Phase

So that you update for cases where you transition to `Running`

Reply:
Thanks for mentioning, fixed!